src/main/antlr/bio/terra/grammar/SQL.g4

```diff
@@ -0,0 +1,387 @@
+grammar SQL;
+
+@header {
+    package bio.terra.grammar;
+}
+
+// includes EOF in order to match the entire input
+query_statement: query_expr EOF;
+
+query_expr: select_statement limit_clause?;
+
+select_statement: SELECT (ALL | DISTINCT)?
+    ( ( expr? '.'? '*' (except_statement)? ) | expr (AS? alias_name)? ) ( ',' ( ( expr?  '*' (except_statement)? ) | expr (AS? alias_name)? ) )*
+    from_statement?
+    where_statement?;
+
+from_statement : FROM from_item (',' from_item )* ;
+
+from_item : table_expr (AS? alias_name)?  (FOR SYSTEM TIME AS OF string)?
+    | from_item join_type? JOIN from_item (on_clause | using_clause)
+    | '(' query_statement ')' (AS? alias_name)?
+    | UNNEST'(' array_expr ')' (AS? alias_name)? (WITH OFFSET (AS? alias_name))?
+    ;
+
+where_statement : WHERE bool_expression;
+
+limit_clause : LIMIT count;
```

Comment:
Either way, I would say that the full capabilities provided by SQL should be out of scope for this particular PR as it is limited to a restricted workflow supporting filtering data for snapshots in the UI using a query to return a subset of results instead of providing a long list of rowids (although certainly we want to be flexible with supporting additional capabilities in the future).